protobuf/builtin.go

```diff
@@ -0,0 +1,13 @@
+package protobuf
+
+func newBuiltin(s string) Builtin {
```

Comment:
Think it's OK to bring the `Builtin` type to this file to keep it next to the rest of his friends?

Reply:
@jprobinson That's just my organization, so I can do either way. You can tell me follow yours and I'll do it, but here's how I do it:

As projects grow, I find that having type definitions scattered in different files confusing -- and I really like the C-header approach where you look at the .h files and you can get the idea of an API set.

In my case I shove most non-local declarations to the `interface.go` file.

Let me know what you think, and I'll change this if need be.